cordapp-example/workflows-kotlin/src/test/kotlin/com/example/test/flow/IOUPaybackFlowTests.kt

```diff
@@ -0,0 +1,93 @@
+package com.example.test.flow
+
+import com.example.flow.ExampleFlow
+import com.example.flow.PaybackFlow
+import com.example.state.IOUState
+import net.corda.core.transactions.SignedTransaction
+import net.corda.core.utilities.getOrThrow
+import net.corda.testing.core.singleIdentity
+import net.corda.testing.node.MockNetwork
+import net.corda.testing.node.MockNetworkParameters
+import net.corda.testing.node.StartedMockNode
+import net.corda.testing.node.TestCordapp
+import org.junit.After
+import org.junit.Before
+import org.junit.Ignore
+import org.junit.Test
+import kotlin.test.assertFailsWith
+
+class IOUPaybackFlowTests {
+    private lateinit var network: MockNetwork
+    private lateinit var lender: StartedMockNode
+    private lateinit var borrower: StartedMockNode
+    private lateinit var iouInput: IOUState
+
+    // create input IOU - we need input for flow test
+    private fun issueIOU(): IOUState {
+        val flow = ExampleFlow.Initiator(9, borrower.info.singleIdentity())
+        val future = lender.startFlow(flow)
+        network.runNetwork()
+
+        val stx: SignedTransaction? = future.get()
+        return stx?.tx?.outputs?.get(0)?.data as IOUState
```

Comment:
@xavierlepretre @adelRestom  - this is one thing I feee I need to resolve in this exercise.
When I wanted to test a payback flow, I need to create input. But when I'm doing so, I'm getting error for CashContract:
```
"Creator must be a signer" using (command.signers.single() == cashState.owner.owningKey)
```
I have unit tests for all contract constraints so it should be ok.

could you please take a look? What I doing wrong here?

Reply:
In `ExampleFlow`:

```kotlin
val cashState = CashState(iouState.value.toLong(), iouState.borrower)
val iouCommand = Command(IOUContract.Commands.Create(), iouState.participants.map { it.owningKey })
val cashCommand = Command(CashContract.Commands.Create(), ourIdentity.owningKey)
```
`ourIdentity` is the lender, but the cash state owner is `iouState.borrower`...

Reply:
When you solve the issuer / owner thing, it will work out.